### ISSUE TITLE
Access for Jeffrey Wallace (elevated permissions)

### ISSUE THREAD

**Issue by jw81** [department-of-veterans-affairs/va.gov-team#37403]
I'm the lead developer on the Benefits Claims API in Lighthouse.  I need to look at some Prod records of ours and find records that meet certain criteria.  (see details below)  Depending on what we find, we may need to manually re-run some background logic for those records. (see details below)

Besides this specific occurence, it would also be helpful to have some form of Prod access in order to help in troubleshooting future issues.  I understand Prod access is a delicate matter though, so I want to make sure I identify and follow all the proper processes first.

Initially, I would just like to identify what records meet this criteria ::
```
# Find all records with specific error message
failed_records = ClaimsApi::PowerOfAttorney.where(vbms_error_message: 'VBMS is unable to locate file number')
```

Depending on what we find and after communicating with the affected consumer, we might potentially want to run this ::
```
# Re-run logic for uploading attachment to VBMS
failed_records.each do |record|
  ClaimsApi::VBMSUploadJob.perform_async(record.id)
end
```

**Comment by jsgarmon** [comment]
@department-of-veterans-affairs/operations-aws-approvers : I would treat this as an access request for a prod resource. Similar to other cases, on your approval we would grant the user the temporary ability to start SSM sessions in production

**Comment by jw81** [comment]
> 

@jsgarmon Would that be pretty much the same thing I already do when I need to launch a console in Sandbox?
I currently run this `sh ./ssm.sh vets-api-server sandbox`.  I'm guessing maybe it would be the same process, but passing it `production` instead?

**Comment by jsgarmon** [comment]
yep, that should do it

**Comment by jw81** [comment]
@jsgarmon it looks like this ticket is in an "Awaiting Approval" status.
Anything I can do to help move it along?

**Comment by jsgarmon** [comment]
any of @department-of-veterans-affairs/operations-aws-approvers can grant approval. Typically @td-usds reviews production access requests.

**Comment by jw81** [comment]
I still cannot access a Rails console in Prod.
I get this error ::
```
An error occurred (AccessDeniedException) when calling the StartSession operation: User: arn:aws-us-gov:iam::REMOVED-JUST-IN-CASE:user/Jeff.Wallace is not authorized to perform: ssm:StartSession on resource: arn:aws-us-gov:ec2:us-gov-west-1:REMOVED-JUST-IN-CASE:instance/i-REMOVED-JUST-IN-CASE because no identity-based policy allows the ssm:StartSession action
```

Is there anyone in @department-of-veterans-affairs/operations-aws-approvers or @td-usds who can assist me?  

This ticket has been open for 2 weeks.

**Comment by td-usds** [comment]
If I understand correctly, you need to:
1. run some exploratory queries
2. depending on what those turn up, run 'some' `insert`s, `update`s, and/or `delete`s

Is that accurate?
I am _very_ uncomfortable with the whole Rails Console access so I'm wondering if there's another thing we can do:
What if we give you a _copy_ of the prod database that you can psql into to your hearts content. That will let you do the exploratory querying and will let you formulate the CUD statements you'll need infra to apply; would that work?

**Comment by jw81** [comment]
> If I understand correctly, you need to:
> 
> 1. run some exploratory queries
> 2. depending on what those turn up, run 'some' `insert`s, `update`s, and/or `delete`s
> 
> Is that accurate? I am _very_ uncomfortable with the whole Rails Console access so I'm wondering if there's another thing we can do: What if we give you secure access to a _copy_ of the prod database that you can psql into to your hearts content. That will let you do the exploratory querying and will let you formulate the CUD statements you'll need infra to apply; would that work?

Unfortunately, our need for access to Prod isn't going to end once we're past this first hurdle. We have many systems we have to interact with and most times there's no way to investigate and diagnose where the problems are without being able to run code snippets to see what is happening.

**Comment by td-usds** [comment]
This is exactly what gives me great concern:

> without being able to run code snippets to see what is happening.

This sounds like experimenting in prod to me.

Is there a way to kind of "package up" what would have to be done on the prod system to correct the issue so that it can be reviewed and signed off on as "something to unleash upon production" rather than literally doing it live in prod with all the potential for badness to happen there? That's the thing I'm really looking for...
In other words: what are all the systems & data that you'd need access for and that we could maybe create/make available in a non-prod environment for you so that you _can_ do the exploration in a safe manner, while ensuring that the output of that exploration can result in a reviewable script to then apply that set of instructions (whatever they are) to prod.

**Comment by td-usds** [comment]
@jw81 Is this still an active issue? If so, can you provide more information about which exact things you need access to?

**Comment by td-usds** [comment]
ping @jw81

**Comment by jbritt1** [comment]
Over 30 days stale / awaiting reply from author, closing. Please re-open if / when we need to revisit the topic.

**Comment by jw81** [comment]
@jbritt1 @mchelen-gov 
My apologies.  I did not see the prior attempts to ping me until just now.

Basically, what's going on is my API accepts Veteran's disability claims.  We take that data and then send it to EVSS so that they can establish a Veteran's disability claim with the VA.  When the submission to EVSS fails in prod because the data is invalid in some way, we have no way of replicating the problem in a lower environment because we can't see the invalid data that was given to us.  So we need the ability to hop into prod and look at the data so that we can reproduce the problem and figure out how to fix it.

Let me know if a Slack thread would be a better/quicker way to talk through the need.

**Comment by jbritt1** [comment]
Think the PO you want to tag here would be @td-usds

**Comment by SlakrHakr** [comment]
Can we reopen this to get it moving forward again? or does a new issue need to be created and linked to this one?

**Comment by jbritt1** [comment]
Let's go the route of the proper AWS access template please.

Here is the template: https://github.com/department-of-veterans-affairs/va.gov-team/issues/new?assignees=&labels=external-request%2Coperations%2Cops-access-request&template=aws-access-request.yml&title=AWS+access+for+%5Bindividual%5D
Here is an example: https://github.com/department-of-veterans-affairs/va.gov-team/issues/42776

Sorry for the switch-up, was hoping for more input. I feel the AWS template is the more correct path forwards because it requests some additional info that will be needed like COR/VOR info.

**Comment by dabaranowski** [comment]
@SlakrHakr @jbritt1 thanks for getting this moving forward. @jw81 let me know if you need anything from me for the new process.

**Comment by jw81** [comment]
@jbritt1 @dabaranowski 
I believe I submitted one of those templates 2 weeks ago. 

https://github.com/department-of-veterans-affairs/va.gov-team/issues/42885

**Comment by jbritt1** [comment]
> @jbritt1 @dabaranowski I believe I submitted one of those templates 2 weeks ago.
> 
> #42885

There is also a comment there as to why it has not yet moved forward https://github.com/department-of-veterans-affairs/va.gov-team/issues/42885#issuecomment-1156469221

**Comment by jw81** [comment]
Our PM was trying to track down what the roster was because she wasn't familiar with it and our PO was OOO for a couple weeks but she was fairly certain he didn't know about the roster either.  We'll try to figure out what the roster is and how to add to it.

**Comment by jbritt1** [comment]
In the comment there is a link to the platform orientation guide, from there it will lead you to more instructions that essentially lands you at https://github.com/department-of-veterans-affairs/va.gov-team/issues/new?assignees=&labels=governance-team%2C+analytics-insights%2C+analytics-request&template=orientation-epic.md&title=Platform+Orientation+Template+%5BYour+name+here%5D. We are aware you are not new to the platform and this new rule is perhaps a little confusing. Again though, it was decided at the PO level and out of our hands.

**Comment by pshahVA** [comment]
Ok, we got the roster updated. What are the next steps?
